server/src/main/java/io/crate/rest/action/SqlHttpHandler.java

```diff
@@ -294,12 +296,21 @@ private CompletableFuture<XContentBuilder> executeBulkRequest(Session session,
     }
 
     Role userFromAuthHeader(@Nullable String authHeaderValue) {
-        String username = Headers.extractCredentialsFromHttpBasicAuthHeader(authHeaderValue).v1();
-        // Fallback to trusted user from configuration
-        if (username == null || username.isEmpty()) {
-            username = AuthSettings.AUTH_TRUST_HTTP_DEFAULT_HEADER.get(settings);
+        Tuple<String, SecureString> credentials = null;
```

Comment:
Maybe use a private record, as I am under the assumption we try to get rid of tuple.

Reply:
There is https://github.com/crate/crate/pull/15497 for that. 
It will be a Credentials class because password is set later on in PG - https://github.com/crate/crate/pull/15497#discussion_r1475806372

Reply:
Updated to tuple - took some bits of https://github.com/crate/crate/pull/15497